src/commands/publish.ts

```diff
@@ -0,0 +1,77 @@
+import * as path from "path";
+import cli from "cli-ux";
+
+import TartCommand from "../TartCommand";
+import { createExecaCommand, parseGitOutput, GIT_FLAGS } from "../utils";
+
+export default class Publish extends TartCommand {
+  static description = "publish specified dump file to a remote repository";
+
+  static examples = [`$ tart publish filename`];
+
+  static flags = {
+    ...TartCommand.flags,
+  };
+
+  static args = [
+    {
+      name: "file",
+      required: true,
+      description: "name of the file",
+    },
+  ];
+
+  async run() {
+    await this.runHook("beforePublish");
+
+    const { args } = this.parse(Publish);
+    const { file } = args;
+    const url = this.localConfig.repository;
+
+    this.log(`file: ${file} || repo: ${url}`);
+
+    const git = createExecaCommand("git", {
+      cwd: this.localConfig.repoDir,
+    });
+
+    // check that tag exists locally
+    const { stdout } = await git(["tag"]);
+    if (!stdout.includes(file)) {
+      this.log(
+        `${file} is not a valid file in the repo. Try:\n   tart save ${file}@\n   tart publish ${file}@`
+      );
+      return;
+    }
+
+    // try git push <repo url> <tag name>
+    try {
+      const { stderr } = await git(["push", "--verbose", url, file]);
+
+      const flag = parseGitOutput(stderr);
+      if (flag === GIT_FLAGS.UP_TO_DATE) {
+        this.log("File is up to date in remote repo.");
+      } else if (flag === GIT_FLAGS.ADD) {
+        this.log("Success, file added to remote repo.");
+      }
+    } catch (err) {
```

Comment:
we might want to have an else that does this.error, otherwise the caught error will be eaten